orc8r/cloud/go/services/orchestrator/obsidian/handlers/gateway_handlers.go

```diff
@@ -11,69 +11,66 @@
 
+// File gateway_handlers.go provides generic gateway handlers, with hooks for
+// specific gateway types.
+//
+// These handlers do not support updating a gateway's ID.
+
 package handlers
 
 import (
+	"fmt"
 	"net/http"
 
+	"github.com/go-openapi/swag"
+
 	"magma/orc8r/cloud/go/obsidian"
 	"magma/orc8r/cloud/go/orc8r"
 	"magma/orc8r/cloud/go/serde"
 	"magma/orc8r/cloud/go/services/configurator"
 	"magma/orc8r/cloud/go/services/device"
 	"magma/orc8r/cloud/go/services/orchestrator/obsidian/models"
-	"magma/orc8r/cloud/go/services/state"
+	"magma/orc8r/cloud/go/services/state/wrappers"
 	"magma/orc8r/cloud/go/storage"
 	merrors "magma/orc8r/lib/go/errors"
 
-	"github.com/go-openapi/swag"
 	"github.com/labstack/echo"
 	"github.com/pkg/errors"
 )
 
-// MagmadEncompassingGateway is an interface for a gateway API model which
-// wraps the magmad gateway with more fields that translate into additional
-// network entities in the storage layer.
-// Note that *models.MagmadGateway itself implements this interface as well.
-type MagmadEncompassingGateway interface {
-	// ValidatableModel allows the model to be validated by calling
-	// ValidateModel()
+// GatewaySubtype represents a subtype of the Magmad gateway.
+// The gateway subtype wraps the base Magmad gateway with additional fields by
+// creating and associating further network entities with the Magmad gateway.
+//
+// Note: since the default Magmad gateway model implements this interface as
+// well, DO NOT return the base Magmad model from any of the "get additional"
+// methods.
+type GatewaySubtype interface {
 	serde.ValidatableModel
 
-	// GetMagmadGateway returns the *models.MagmadGateway which is wrapped by
-	// the model
+	// GetMagmadGateway returns the Magmad gateways wrapped by the subtype.
 	GetMagmadGateway() *models.MagmadGateway
 
 	// GetAdditionalWritesOnCreate returns extra write operations to perform
-	// during creation, inside the same backend transaction as the creation
-	// of the magmad gateway.
-	// Do NOT include the write operation for the magmad gateway in the return
-	// value, as *models.MagmadGateway itself implements this interface and
-	// will create itself.
+	// during creation.
+	// The writes are performed in the same backend transaction with creation
+	// of the Magmad gateway.
 	GetAdditionalWritesOnCreate() []configurator.EntityWriteOperation
 
-	// GetAdditionalEntitiesToLoadOnUpdate is a **static** method which
-	// returns a list of entity keys to load in addition to the magmad gateway
-	// during an update operation. The gateway ID from the API URL parameter
-	// is given as an argument.
+	// GetAdditionalLoadsOnUpdate returns a list of additional entity keys to
+	// load during an update.
 	// The entities loaded during this operation will be passed to
-	// `GetAdditionalWritesOnUpdate`.
-	GetAdditionalEntitiesToLoadOnUpdate(gatewayID string) []storage.TypeAndKey
+	// GetAdditionalWritesOnUpdate.
+	GetAdditionalLoadsOnUpdate() []storage.TypeAndKey
```

Comment:
Why did you choose to make this non-static? Do we always have access to the gateway ID on a bound model?

Reply:
Yep, it's only called in `UpdateGateway`, where the model is generated from the payload